docs/data/data-grid/server-side-data/ServerSideLazyLoadingViewport.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  GridDataSource,
+  GridGetRowsParams,
+} from '@mui/x-data-grid-pro';
+import { useMockServer } from '@mui/x-data-grid-generator';
+
+function ServerSideLazyLoadingViewport() {
+  const { columns, fetchRows } = useMockServer(
+    { rowLength: 100 },
+    { useCursorPagination: false, minDelay: 200, maxDelay: 500 },
+  );
+
+  const dataSource: GridDataSource = React.useMemo(
+    () => ({
+      getRows: async (params: GridGetRowsParams) => {
+        const urlParams = new URLSearchParams({
+          filterModel: JSON.stringify(params.filterModel),
+          sortModel: JSON.stringify(params.sortModel),
+          firstRowToRender: `${params.start}`,
+          lastRowToRender: `${params.end}`,
+        });
+        const getRowsResponse = await fetchRows(
+          `https://mui.com/x/api/data-grid?${urlParams.toString()}`,
+        );
+
+        return {
+          rows: getRowsResponse.rows,
+          rowCount: getRowsResponse.rowCount,
+        };
+      },
+    }),
+    [fetchRows],
+  );
+
+  return (
+    <div style={{ width: '100%', height: 400 }}>
+      <DataGridPro
+        columns={columns}
+        unstable_dataSource={dataSource}
+        lazyLoading
+        paginationModel={{ page: 0, pageSize: 10 }}
```

Comment:
@arminmeh Should we make chunk size independent to `pageSize` to have https://github.com/mui/mui-x/pull/13878#discussion_r1797314764 work even better?

Let's say the `pageSize` is `10` but chunkSize could be `50` resulting in fetching 5 pages in one request (no need to change `pageSize` to see the effect, the Grid applies the updates in the order of `pageSize`, 10 at a time in this case).

This would also allow us to apply the same optimization to other cases too.
Examples: paginated server-side data, row grouped/tree data, [nested lazy loading](https://github.com/mui/mui-x/issues/14527) [future], etc.

I think AgGrid also [follows](https://www.ag-grid.com/javascript-data-grid/server-side-model-configuration/) a similar route (they call it `cacheBlockSize`). It opens up the possibility of optimizing the backend fetching per server/project needs.

Wdyt?

Reply:
Maybe I need a better naming :)
The goal of chunk manager (let's call it that) is to increase the amount of cache hits.

If the user scrolls to fast, rendering context increases and lazy loading tries to fetch more pages upfront in one request (to reduce the amount of requests and to fill in more data as you are scrolling fast through the rows).

Current implementation takes the data that you are about to cache, calculates minimum possible page size and splits the results into those pieces. Later, if you do sorting/filtering, clear the current rows and come back with the same params but different rendering context, chunk manager will be able to find smaller piece of the original fetch result.

Same goes the other way around. If you suddenly start scrolling fast through pages that you already saw, chunk manager will concatenate those records into one array.

I like your idea though and we should make it, but I would call that `prefetching`. We can allow configuration to do it in multiple directions like prefetch few pages upfront, prefetch certain sorting model or filter model states after initial load. Other possibility is to prefetch on hover (before sorting is initiated), etc...

But, this is definitely something for another issue/PR

Reply:
> Does the pageSize represent the chunk size? It would be great to have it covered in the docs.

Documented here
https://deploy-preview-13878--material-ui-x.netlify.app/x/react-data-grid/server-side-data/#improving-the-cache-hit-ratio

I have created a new issue for [preloading](https://github.com/mui/mui-x/issues/15611)